src/commands/upload.rs

```diff
@@ -781,12 +813,54 @@ pub async fn upload_submission(
         } else {
             "You must agree to the competition rules before submitting."
         };
-        let mut url = global.aqora_url().unwrap();
-        url.set_path(&format!("competitions/{slug}/rules"));
-        return Err(error::user(
-            message,
-            &format!("Please agree to the competition rules at {url}",),
-        ));
+        let mut rules = DEFAULT_RULES.to_string();
+        if !rule_text.trim().is_empty() {
+            rules.push_str(&format!("\n\n{rule_text}"));
+        }
+
+        let accepts = m.suspend(|| {
+            let will_review = dialoguer::Confirm::with_theme(global.color.dialoguer().as_ref())
+                .with_prompt(format!("{message} Would you like to review them now?"))
+                .default(true)
+                .interact()
+                .ok()
+                .unwrap_or_default();
+            if !will_review {
+                return false;
+            }
+            if dialoguer::Editor::new().edit(&rules).is_err() {
```

Comment:
I think we should have used `$PAGER` here, then fallback on `$VISUAL` and then `$EDITOR`, then `vi`. But `dialoguer::Editor` is available right now out of the box so we can get back here later